src/main/java/kr/syeyoung/dungeonsguide/mod/DungeonsGuide.java

```diff
@@ -0,0 +1,303 @@
+    }
+
+    private boolean showedStartUpGuide;
+    @SubscribeEvent
+    public void onGuiOpen(GuiOpenEvent guiOpenEvent){
+        if(!showedStartUpGuide){
+            showedStartUpGuide = true;
+
+            if(isFirstTimeUsingDG()){
+                GuiScreen originalGUI = guiOpenEvent.gui;
+                guiOpenEvent.gui = new GuiScreen() {
+                    final String welcomeText = "Thank you for installing §eDungeonsGuide§f, the most intelligent skyblock dungeon mod!\nThe gui for relocating GUI Elements and enabling or disabling features can be opened by typing §e/dg\nType §e/dg help §fto view full list of commands offered by dungeons guide!";
+
+                    @Override
+                    public void initGui() {
+                        ScaledResolution sr = new ScaledResolution(Minecraft.getMinecraft());
+                        this.buttonList.add(new GuiButton(0, sr.getScaledWidth()/2-100,sr.getScaledHeight()-70 ,"Continue"));
+                    }
+
+                    @Override
+                    protected void actionPerformed(GuiButton button) throws IOException {
+                        super.actionPerformed(button);
+                        if (button.id == 0) {
+                            Minecraft.getMinecraft().displayGuiScreen(originalGUI);
+                        }
+                    }
+
+                    @Override
+                    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+                        super.drawBackground(1);
+
+                        ScaledResolution sr = new ScaledResolution(Minecraft.getMinecraft());
+                        FontRenderer fontRenderer = Minecraft.getMinecraft().fontRendererObj;
+                        fontRenderer.drawString("§eWelcome To DungeonsGuide", (sr.getScaledWidth()-fontRenderer.getStringWidth("Welcome To DungeonsGuide"))/2,40,0xFFFF0000);
+                        int tenth = sr.getScaledWidth() / 10;
+                        Gui.drawRect(tenth, 70,sr.getScaledWidth()-tenth, sr.getScaledHeight()-80, 0xFF5B5B5B);
+
+                        String[] split = welcomeText.split("\n");
+                        for (int i = 0; i < split.length; i++) {
+                            fontRenderer.drawString(split[i].replace("\t", "    "), tenth + 2,i*fontRenderer.FONT_HEIGHT + 72, 0xFFFFFFFF);
+                        }
+
+                        super.drawScreen(mouseX, mouseY, partialTicks);
+                    }
+
+                };
+            }
+
+        }
+    }
+
+
+    public void preinit(){
+
+        String version = null;
+        try (InputStream resourceAsStream = this.getClass().getResourceAsStream("/kr/syeyoung/dungeonsguide/DungeonsGuide.class")) {
+            if (resourceAsStream == null) {
+                if (System.getProperty("dg.version") == null) {
+                    version = "nlatest";
+                } else {
```

Comment:
Pretty sure this goes into Main.java

Supporter auto-update jar only has Main, GuiLoadingErr, IDG, and auth stuff

it's Main's job to download DungeonsGuide from my server, then register it into url and do some hax to dynamically load it

Soooo this code never runs in case when this code needs to be ran